e2e/imgbuild/build_utils.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package imgbuild
+
+import (
+	"io/ioutil"
+	"log"
+	"os/exec"
+	"path"
+	"strings"
+	"testing"
+	"text/template"
+
+	"github.com/sylabs/singularity/e2e/actions"
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+// Opts define image build options
+type Opts struct {
+	Force   bool
+	Sandbox bool
+	Env     []string
+}
+
+type defFileDetail struct {
+	Bootstrap string
+	From      string
+	Registry  string
+	Namespace string
+	Labels    map[string]string
+}
+
+// ImageBuild builds an image based on the Opts
```

Comment:
I same as before. I think having the options hard coded could be bad. But I think the wrappers I suggested would be good inside helper functions like this `ImageBuild`